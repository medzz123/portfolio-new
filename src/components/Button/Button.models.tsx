import { ButtonHTMLAttributes } from 'react';

export type ButtonProps = ButtonHTMLAttributes<HTMLButtonElement> & {
  busy?: boolean;
};
