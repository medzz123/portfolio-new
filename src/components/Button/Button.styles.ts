import { styled } from '@theme/config';

export const ButtonContainer = styled('button', {
  padding: '$10 $28',
  boxShadow: 'none',
  border: 'none',
  backgroundColor: '$primary',
  borderRadius: '$sizes$12',
  color: '$loContrast',
  fontSize: '$info',
  fontWeight: '$normal',
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center',
  cursor: 'pointer',
  '&:hover, &:focus': {
    boxShadow: '0 0 0 $sizes$2 $colors$success',
  },
  '&:disabled': {
    backgroundColor: '$grey800',
    cursor: 'default',
  },
  position: 'relative',
});

export const TextContainer = styled('span', {
  transition: 'all 200ms ease',
  variants: {
    hide: {
      hidden: {
        opacity: 0,
      },
      visible: {
        opacity: 1,
      },
    },
  },
});

export const SpinnerContainer = styled('div', {
  position: 'absolute',
  left: '$half',
  top: '$half',
  transform: 'translate(-$sizes$half, -$sizes$half)',
  svg: {
    width: '$20',
    height: '$20',
  },
});
