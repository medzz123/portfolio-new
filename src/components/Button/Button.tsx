import Spinner from '@components/Spinner';
import React, { FunctionComponent } from 'react';

import { ButtonProps } from './Button.models';
import { ButtonContainer, SpinnerContainer, TextContainer } from './Button.styles';

const Button: FunctionComponent<ButtonProps> = (props) => {
  const { type = 'button', children, busy = false, ...rest } = props;

  return (
    // @ts-ignore
    <ButtonContainer type={type} disabled={busy} data-testid="button" {...rest}>
      <TextContainer hide={busy ? 'hidden' : 'visible'}>{children}</TextContainer>
      {busy && (
        <SpinnerContainer>
          <Spinner />
        </SpinnerContainer>
      )}
    </ButtonContainer>
  );
};

export default Button;
