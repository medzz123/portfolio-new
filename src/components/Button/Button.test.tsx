import { render } from '@test/library';
import React from 'react';

import Button from './Button';

test('Button Test', () => {
  const { getByTestId, container } = render(<Button>Hello</Button>);

  const buttonNode = getByTestId('button');

  expect(buttonNode.textContent).toBe('Hello');

  expect(container).toMatchSnapshot();
});
