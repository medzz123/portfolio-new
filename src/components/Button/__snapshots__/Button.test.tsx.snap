// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Test 1`] = `
<div>
  <button
    class="sxexqz4"
    data-testid="button"
    type="button"
  >
    <span
      class="sxmjyut sxmjyut8iuhr--hide-visible"
    >
      Hello
    </span>
  </button>
</div>
`;
